modules/assim.sequential/R/ensemble_downscale.R

```diff
@@ -0,0 +1,266 @@
+##' @title Subset ensemble data for downscaling
+##' @name SDA_downscale_preprocess
+##' @author Sambhav Dixit, David LeBauer
+##'
+##' @param ensemble_data EFI standard tibble or data.frame
+##' @param site_coords data.frame with unique site id
+##' @param date Date. The date for the run, must be a date within `ensemble_data`.
+##' @param carbon_pool Character. Carbon pool of interest. Name must match the carbon pool name in ensemble_data. 
+##' found within the file or object supplied to 'ensemble_data'.
+##' @details This function subsets ensemble data and ensures that the specified date and 
+##' carbon pool are present in the ensemble data.
+##'
+##' @return A list containing the cleaned site coordinates and the ensemble carbon output for the 
+##' specified date and carbon pool.
+##'
+##' @export
+
+subset_ensemble <- function(ensemble_data, site_coords, date, carbon_pool) {
+
+  # Confirm date is in ensemble data
+  if (!any(lubridate::date(unique(ensemble_data$datetime)) == lubridate::date(date))) {
+    PEcAn.logger::logger.error(paste(
+      "Provided date", date,
+      "is not found in the ensemble_data input."
+    ))
+  }
+
+  # Ensure the carbon pool exists in the input data
+  if (!carbon_pool %in% unique(ensemble_data$variable)) {
+    PEcAn.logger::logger.error("Carbon pool", carbon_pool, "not found in the input data.")
+  }
+
+  # Ensure the sites are in the ensemble data
+  if (!all(unique(site_coords$id) %in% unique(ensemble_data$site_id))) {
+    PEcAn.logger::logger.error("Some sites in site_coords are not present in the ensemble_data.")
+  }
+
+  # Filter the ensemble data to the specified date and carbon pool
+  ensemble_data <- ensemble_data |>
+    filter(
+      lubridate::date(datetime) == lubridate::date(date),
+      site_id %in% unique(site_coords$id),
+      variable == carbon_pool
+    ) |>
+    select(site_id, ensemble, prediction)  # use site_id instead of site
```

Comment:
```suggestion
    dplyr::select("site_id", "ensemble", "prediction")  # use site_id instead of site
```
not sure what the comment means -- is it needed?